client/src/App.js

```diff
@@ -0,0 +1,125 @@
+import './App.css';
+import { useState } from 'react';
+import Score from './components/Score';
+import Question from './components/Question';
+import Flag from './components/Flag';
+
+function App() {
+
+  const [submitted, setSubmitted] = useState(false);
+  const [questionChosen, setQuestionChosen] = useState(null);
+  const [questionIndex, setQuestionIndex] = useState(0);
+  const [displayQuestion, setDisplayQuestion] = useState(false);
+  const [resultMessage, setResultMessage] = useState(null);
+  const [score, setScore] = useState(0);
+  const [disable, setDisable] = useState(false);
+  const [flagIndex, setFlagIndex] = useState(1);
+
+  const message = [
+    `Sorry, your answer is incorrect.`,
+    `Bravo! Your answer is correct. You are one step closer to completing the puzzle!`, 
+    `Bravo! Your answer is correct. Great job completing parts of the puzzle!`,
+    `Bravo! Your answer is correct. Congratulations! You have successfully completed the puzzle!`
+  ];
+
+  const flagPieces = document.getElementsByClassName('flag');
+
+  const generateQuestion = () => {
+    continueGame();
+    setDisable(false);
+    resetCheckedAnswers();
+    setSubmitted(false);
+    setResultMessage(false);
+    fetchQuestion();
+    setDisplayQuestion(true);
+  }
+
+  const continueGame = () => {
+    // Reset game after all ten questions
+    if (questionIndex === 9) {
+      setQuestionIndex(null);
+    } else if (questionIndex === null) {
+      setScore(0);
+      setQuestionIndex(0);
+      setFlagIndex(1);
+      document.getElementById('button').textContent = `Start Quiz`;
+      for (let flagPiece of flagPieces) {
+        flagPiece.style.visibility = "hidden";
+      }
+    } 
+    // Continue game if not at the end of all questions
+    else {
+      setQuestionIndex(questionIndex => questionIndex + 1)
+    }
+  }
+
+  const resetCheckedAnswers = () => {
+    if (document.getElementById('choice1')) {
+      document.getElementById('choice1').checked = false;
+      document.getElementById('choice2').checked = false;
+      document.getElementById('choice3').checked = false;
+      document.getElementById('choice4').checked = false;
+    }
+  };
+
+  const fetchQuestion = async () => {
+    let response = await fetch('/api');
+    await response.json().then(questions => setQuestionChosen(() => {
+      return questions[questionIndex];
+    }));
+  }
+
+  const handleSubmit = (e) => {
```

Comment:
Again `e` is a confusing variable name. I might try to use a variable name which captures what type of event this is.